### ISSUE TITLE
Marathon remains "Deploying" status when container has already deployed

### ISSUE THREAD

**Issue by kiwenlau** [mesosphere/marathon#4028]
When I deploy nginx with following JSON definition:

```
{
    "id": "nginx",
    "labels": {
        "HAPROXY_GROUP": "external"
    },
    "cpus": 0.2,
    "mem": 20.0,
    "instances": 2,
    "healthChecks": [{
        "path": "/"
    }],
    "container": {
        "type": "DOCKER",
        "docker": {
            "image": "nginx",
            "network": "BRIDGE",
            "portMappings": [{ "containerPort": 80, "hostPort": 0, "servicePort": 80, "protocol": "tcp" }]
        }
    }
}
```

2 instance of nginx containers start immediately because I have pulled nginx image. I can find that nginx containers are running and I can visit nginx through browser. However, the marathon UI shows that the nginx application is still "Deploying". After about **60 seconds**, its status change to "RUNNING".

I found this is related to **healthChecks**, more specifically, the  **intervalSeconds** of **healthChecks**:
- intervalSeconds (Optional. Default: 60): Number of seconds to wait between health checks

So I set **intervalSeconds** to 5:

```
{
    "id": "nginx",
    "labels": {
        "HAPROXY_GROUP": "external"
    },
    "cpus": 0.2,
    "mem": 20.0,
    "instances": 2,
    "healthChecks": [{
        "path": "/",
        "intervalSeconds": 5
    }],
    "container": {
        "type": "DOCKER",
        "docker": {
            "image": "nginx",
            "network": "BRIDGE",
            "portMappings": [{ "containerPort": 80, "hostPort": 0, "servicePort": 80, "protocol": "tcp" }]
        }
    }
}
```

Then I can find that the status of nginx application change to "RUNNING" in about 5 seconds.

However, this is not a good solution. Because when the application is running, we don't need to check the healthy in every 5 seconds. We only need to check the healthy with high frequency when we first start the application.

**Comment by kensipe** [comment]
Thanks for the report @kiwenlau I've looked into this and here is a quick report:

There are 2 things invoked here health and deployment.     From a health perspective, there are currently 3 states possible (healthy, unhealthy and unknown).   As you indicated, it is correct that a health check does NOT start until after the `internalSeconds`.  Prior to that, a task can be "deployed" and be running but have an unknown health.  Only after the internalSeconds will health be known.

From a "Status" standpoint (as reflected in the UI), the label Deploying doesn't seem to change from "Deploying" to "Running" until after a health check is received even if the "Running Instances" have transition from "0 of 2" to "2 of 2", whereas with no health check it does immediately.   The fact that the Status (with health checks provided) stays in "Deploying" status until health checks are received is either a bug or more likely is a design decision by the UI.   I've labeled this issue with a "gui" label and lets see if we can get some help from @leemunroe on what he advices.

I would expect that status would go to running with an unknown health until the health information is provided but there could be reasons why that isn't the case.

**Comment by kensipe** [comment]
As a quick confirmation if using dcos with the first json provided ( 60 sec health checks).  

```
dcos marathon app list
ID      MEM  CPUS  TASKS  HEALTH  DEPLOYMENT  CONTAINER  CMD   
/nginx   20  0.2    2/2    0/2      scale       DOCKER   None  
```

**Comment by kiwenlau** [comment]
Thx. I agree that:

As soon as the container is running, the status should change from "Deploying" to "Running", while the health check can remain "unknown" until it get feedback.

**Comment by kensipe** [comment]
@kiwenlau and @leemunroe I've learned more since my initial comment.  It turns out that there is a `DEPLOYMENT` status of `scale` which is returned until the health check is complete when health checks are provided.   This looks like a fairly simple change to resolve.  I'll schedule this fix for the next release `1.3.`.   This is not a UI issue as shown in the dcos commandline output:

```
dcos marathon app list
ID      MEM  CPUS  TASKS  HEALTH  DEPLOYMENT  CONTAINER  CMD   
/nginx   20  0.2    2/2    0/2      scale       DOCKER   None  
```

Details on fixing.   The `preStart` of `HealthCheckActor` invokes `scheduleNextHealthCheck()` which schedules for the internal.   
https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/health/HealthCheckActor.scala#L78-L81

This should be invoke for all checks except for the first.    The first invocation should be immediately scheduled.
https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/health/HealthCheckActor.scala#L36
